### PR TITLE
AVNM Scope Connection APIs

### PR DESCRIPTION
**Pull request by microsoft-miroos**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  September 2021
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  September 2021
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [x] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPendingConnectionList.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "title": "List Pending Azure Virtual Network Manager Scope Connection",
+  "parameters": {
+    "api-version": "2021-02-01-preview",
+    "subscriptionId": "subscriptionC",
+    "resourceGroupName": "rg1",
+    "networkManagerName": "testNetworkManager"
```
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPendingConnectionList.json:7]: >networkManagerName [](http://example.com/codeflow?start=5&length=18)

not needed

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPendingConnectionList.json [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+              "lastModifiedByType": "User",
+              "lastModifiedAt": "2021-01-11T18:52:27Z"
+            },
+            "properties": {
+              "networkManagerId": "/subscriptions/subscriptionC/resourceGroup/rg1/providers/Microsoft.Network/networkManagers/testNetworkManager",
+              "connectionState": "Pending"
+            }
+          }
+        ],
+        "nextLink": "{baseurl}//subscriptions/subscriptionC/resourceGroups/rg1/providers/Microsoft.Network/networkManagers/testNetworkManager/listPendingNetworkManagerConnections?api-version=2021-02-01-preview&$skipToken=10"
```
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPendingConnectionList.json:35]: >listPendingNetworkManagerConnections [](http://example.com/codeflow?start=142&length=36)

Is this correct ? This seems to be defined as a POST call . .but the next link would be a GET

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerScopeConnectionList.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+              "subscriptions": [
+                {
+                  "id": "subscriptionA",
+                  "state": "Pending"
+                }
+              ]
+            }
+          }
+        ],
+        "nextLink": "{baseurl}/subscriptions/subscriptionC/resourceGroup/rg1/providers/Microsoft.Network/networkManagers/testNetworkManager/scopeConnections?api-version=2021-02-01-preview&$skipToken=10"
```
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerScopeConnectionList.json:49]: Same here

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerScopeConnectionList.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{
+  "title": "List Azure Virtual Network Manager Scope Connection",
+  "parameters": {
+    "api-version": "2021-02-01-preview",
+    "subscriptionId": "subscriptionC",
+    "resourceGroupName": "rg1",
+    "networkManagerName": "testNetworkManager"
```
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerScopeConnectionList.json:7]: >networkManagerName [](http://example.com/codeflow?start=5&length=18)

Is this needed ?

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "List Network Manager Connection": {
+            "$ref": "./examples/NetworkManagerConnectionList.json"
+          }
+        }
+      }
+    },
+    "/{scope}/providers/Microsoft.Network/listPendingScopeConnections": {
```
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json:189]: >listPendingScopeConnections [](http://example.com/codeflow?start=42&length=27)

Is the intention here to enalbe the for use this POST action in an arm template ?

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManager.json [hunk 6]
```diff
@@ -507,6 +507,13 @@
                 "type": "string"
               },
               "description": "List of subscriptions."
+            },
+            "crossTenantScopes": {
+              "type": "array",
```
Comment (raosuhas) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManager.json:512]: Please get sign off from breaking changes

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPendingConnectionList.json (hunk 2 above)
Comment (microsoft-miroos) [reply specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/examples/NetworkManagerPendingConnectionList.json:35]: Correct. The listPending should be a Post call to list which returns a skipToken for subsequent pages, not a nextLink. Will update

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json (hunk 5 above)
Comment (microsoft-miroos) [reply specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json:189]: This is strictly a discoverability API. It is used to list any scopeConnections which reference the base scope in the uri. As opposed to the standard GET list scope sconnections api, which will list all scope connections under the parent network manager resource.

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManager.json (hunk 6 above)
Comment (microsoft-miroos) [reply specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManager.json:512]: This is not a breaking change. It is just the addition of a new property

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json [hunk 7]
```diff
@@ -0,0 +1,323 @@
...
+      "description": "List of network manager connections."
+    }
+  },
+  "parameters": {
+    "ScopeParameter": {
+      "name": "scope",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "Scope id for subscription or management group.",
```
Comment (microsoft-miroos) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json:311]: Can you add an example of what the scope would look like. Take a look at the swagger for Policy Assignment. They have a similar construct

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json [hunk 8]
```diff
@@ -0,0 +1,323 @@
...
+        "description": "List all pending scope connections created by a network manager.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ScopeParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./networkManager.json#/parameters/ListTopParameter"
```
Comment (microsoft-miroos) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerConnections.json:204]: ListPending is server-side pagination. It doesn't take query parameter top or skip. We need to send a post body with the optional skip token

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerScopeConnections.json [hunk 9]
```diff
@@ -0,0 +1,378 @@
...
+          "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+        },
+        {
+          "$ref": "./networkManager.json#/parameters/ResourceGroupNameParameter"
+        },
+        {
+          "$ref": "./networkManager.json#/parameters/NetworkManagerNameParameter"
+        },
+        {
+          "$ref": "#/parameters/scopeConnectionName"
```
Comment (microsoft-miroos) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerScopeConnections.json:49]: Current convention seems to be Pascal Case, can you update to ScopeConnectionName

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerScopeConnections.json [hunk 10]
```diff
@@ -0,0 +1,378 @@
...
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "./networkManager.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./networkManager.json#/parameters/NetworkManagerNameParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
```
Comment (microsoft-miroos) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerScopeConnections.json:216]: Same as other side. Post needs a body

File: specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerScopeConnections.json [hunk 11]
```diff
@@ -0,0 +1,378 @@
...
+      "type": "object",
+      "properties": {
+        "tenantId": {
+          "type": "string",
+          "description": "Tenant ID."
+        },
+        "managementGroups": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ScopeConnectionManagementGroup"
```
Comment (microsoft-miroos) [inline specification/network/resource-manager/Microsoft.Network/preview/2021-02-01-preview/networkManagerScopeConnections.json:278]: I don't think these need to be two different objects. Mg and Sub will have the exact same schema. And, the id property is a string anyway. Consolidate into a single object

**Pull request closed without merging**